File: iBurn/src/main/java/com/gaiagps/iburn/database/Art.java
package com.gaiagps.iburn.database;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import android.content.Context;
import androidx.annotation.NonNull;
import android.text.TextUtils;

import com.gaiagps.iburn.AudioTourManager;

import static com.gaiagps.iburn.database.Art.TABLE_NAME;

/**
 * Created by dbro on 6/8/17.
 */

@Entity(tableName = TABLE_NAME)
public class Art extends PlayaItem {
    public static final String TABLE_NAME = "arts";

    public static final String ARTIST = "artist";
    public static final String ARTIST_LOCATION = "a_loc";
    public static final String IMAGE_URL = "i_url";
//    public static final String AUDIO_TOUR_URL = "a_url";


    @ColumnInfo(name = ARTIST)
    public String artist;

    @ColumnInfo(name = ARTIST_LOCATION)
    public String artistLocation;

    @ColumnInfo(name = IMAGE_URL)
    public String imageUrl;

//    @ColumnInfo(name = AUDIO_TOUR_URL)
//    public String audioTourUrl;

    public boolean hasAudioTour(@NonNull Context context) {
//        return !TextUtils.isEmpty(audioTourUrl);
        return AudioTourManager.hasAudioTour(context, playaId);
    }

    public boolean hasImage() {
        return !TextUtils.isEmpty(imageUrl);
    }

}

File: iBurn/src/main/java/com/gaiagps/iburn/database/Art.kt
package com.gaiagps.iburn.database

import android.content.Context
import android.os.Parcelable
import android.text.TextUtils
import androidx.room.ColumnInfo
import androidx.room.Entity
import com.gaiagps.iburn.AudioTourManager
import kotlinx.parcelize.Parcelize

/**
 * Created by dbro on 6/8/17.
 */
@Parcelize
@Entity(tableName = Art.TABLE_NAME)
class Art : PlayaItem(), Parcelable {
    //    public static final String AUDIO_TOUR_URL = "a_url";
    @JvmField
    @ColumnInfo(name = ARTIST)
    var artist: String? = null

    @JvmField
    @ColumnInfo(name = ARTIST_LOCATION)
    var artistLocation: String? = null

    @JvmField
    @ColumnInfo(name = IMAGE_URL)
    var imageUrl: String? = null

    //    @ColumnInfo(name = AUDIO_TOUR_URL)
    //    public String audioTourUrl;
    fun hasAudioTour(context: Context): Boolean {
//        return !TextUtils.isEmpty(audioTourUrl);
        val pId = playaId ?: return false
        return AudioTourManager.hasAudioTour(context, pId)
    }

    fun hasImage(): Boolean {
        return !TextUtils.isEmpty(imageUrl)
    }

    companion object {
        const val TABLE_NAME: String = "arts"

        const val ARTIST: String = "artist"
        const val ARTIST_LOCATION: String = "a_loc"
        const val IMAGE_URL: String = "i_url"
    }
}